### PR TITLE
Fix upgrade test

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

File: pulp_file/tests/upgrade/pre/test_publish.py [hunk 1]
```diff
@@ -78,10 +78,15 @@ def test_all(self):
            repository versions to be published at same time.
         """
         # Step 1
-        for file_content in get_content(self.repo.to_dict())[FILE_CONTENT_NAME]:
-            modify_repo(self.cfg, self.repo.to_dict(), remove_units=[file_content])
+        repo_content = get_content(self.repo.to_dict())[FILE_CONTENT_NAME][:-1]
```
Comment (fao89) [inline pulp_file/tests/upgrade/pre/test_publish.py:81]: Assuring we always have a content

File: pulp_file/tests/upgrade/pre/test_publish.py [hunk 2]
```diff
@@ -78,10 +78,15 @@ def test_all(self):
...
+        for file_content in repo_content:
+            repository_modify_data = RepositoryAddRemoveContent(
+                remove_content_units=[file_content["pulp_href"]]
+            )
+            modify_response = self.repo_api.modify(self.repo.pulp_href, repository_modify_data)
+            monitor_task(modify_response.task)
         version_hrefs = tuple(ver["pulp_href"] for ver in get_versions(self.repo.to_dict()))
-        non_latest = choice(version_hrefs[:-1])
+        non_latest = choice(version_hrefs[1:-1])
```
Comment (fao89) [inline pulp_file/tests/upgrade/pre/test_publish.py:89]: excluding version 0

Comment (fao89) [reply pulp_file/tests/upgrade/pre/test_publish.py:89]: some versions didn't have content, yielding an empty manifest, that's why the failure was intermittent

**Review by dkliban — APPROVED** [review]

**Comment by fao89** [comment]
https://github.com/pulp/pulp_file/pull/533 needs to be merged first!

**Review by dralley — APPROVED** [review]

**Pull request merged**